src/test/ui/union/union-deref.stderr

```diff
@@ -0,0 +1,20 @@
+error: not automatically applying `DerefMut` on `ManuallyDrop` union field
+  --> $DIR/union-deref.rs:14:14
+   |
+LL |     unsafe { u.f.0 = Vec::new() };
+   |              ^^^
+   |
+   = help: writing to this reference calls the destructor for the old value
+   = help: add an explicit `*` if that is desired, or call `ptr::write` to not run the destructor
+
+error: not automatically applying `DerefMut` on `ManuallyDrop` union field
+  --> $DIR/union-deref.rs:18:14
+   |
+LL |     unsafe { u.f.0.0 = Vec::new() };
+   |              ^^^^^^^
```

Comment:
The span for this is slightly wrong, I'd expect it to only cover `u.f.0`. I don't know how this is happening, since the span is right above.

Reply:
I would guess this has to do with the special handling of multiple tuple field accesses in a row. Perhaps 0.0 is a float token.

Reply:
I didn't know there was such special handling.^^ So this would be a problem elsewhere, in the code that sets up the span for these places?